File: Fairyland_Girl in dream/Assets/FairyLand/Scripts/002mainmenu/packageSystem/Item.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
/// <summary>
/// 物品类(中间类)多对多
/// 表示一个物品项
/// 一般一个武平信息类代表一个物品
/// 其实不然：
/// 游戏中：一个人持有多件同一种物品
///         或者说  一个人持有多件  不同等级的  同一种物品
///         这样的话  单一的类就会出现矛盾   又不可能在玩家累中区修改
///         很熟悉：  数据库的  多对多 的设计思想
///         这里设计为：同一类型的要药品放在同一个格子
///                     同一个类型的装备放在不同的格子
/// </summary>
[Serializable]
public class Item {
    //名称
    private string _name;
    //物品类型:装备还是药品（Equip，Drug）
    private ItemType _itemtype;
    //是哪一个物品
    private ItemInformation _itemInfo;
    //物品等级
    private int _level;
    //物品个数
    private int _count;
    //是否已经装备上了
    private bool _dressed=false;

    #region//封装字段
    public int Level
    {
        get
        {
            return _level;
        }

        set
        {
            _level = value;
        }
    }

    public int Count
    {
        get
        {
            return _count;
        }

        set
        {
            _count = value;
        }
    }

    public ItemInformation ItemInfo
    {
        get
        {
            return _itemInfo;
        }

        set
        {
            _itemInfo = value;
        }
    }

    public bool Dressed
    {
        get
        {
            return _dressed;
        }

        set
        {
            _dressed = value;
        }
    }

    public string Name
    {
        get
        {
            return _name;
        }

        set
        {
            _name = value;
        }
    }

    public ItemType Itemtype
    {
        get
        {
            return _itemtype;
        }

        set
        {
            _itemtype = value;
        }
    }
    #endregion
}

File: Hibernate_FairyLand_Database/src/entity/Item.java
package entity;

import Enums.ItemType;

public class Item {
	// 一个物品的ID
	private int _id;
	// 名称
	private String _name;
	// 物品类型:装备还是药品（Equip，Drug）
	private ItemType _itemtype;
	// 是哪一个物品
	private ItemInformation _itemInfo;

	// 物品等级
	private int _level;
	// 物品个数
	private int _count;
	// 是否已经装备上了
	private Boolean _dressed = false;

	public Item() {

	}

	public Item(int _id, int _level, int _count, Boolean _dressed) {
		super();
		this._id = _id;
		this._level = _level;
		this._count = _count;
		this._dressed = _dressed;
	}

	
	public String get_name() {
		return _name;
	}

	public void set_name(String _name) {
		this._name = _name;
	}

	public ItemType get_itemtype() {
		return _itemtype;
	}

	public void set_itemtype(ItemType _itemtype) {
		this._itemtype = _itemtype;
	}

	public int get_id() {
		return _id;
	}

	public void set_id(int _id) {
		this._id = _id;
	}

	public ItemInformation get_itemInfo() {
		return _itemInfo;
	}

	public void set_itemInfo(ItemInformation _itemInfo) {
		this._itemInfo = _itemInfo;
	}

	public int get_level() {
		return _level;
	}

	public void set_level(int _level) {
		this._level = _level;
	}

	public int get_count() {
		return _count;
	}

	public void set_count(int _count) {
		this._count = _count;
	}

	public Boolean get_dressed() {
		return _dressed;
	}

	public void set_dressed(Boolean _dressed) {
		this._dressed = _dressed;
	}

}
